Add deacon as host removal tool (#757)
nf-core/taxprofiler: docs/usage.md

Similarly to complexity filtering, host-removal can be useful for runtime optimisation and reduction in misclassified reads. It is not always necessary to report classification of reads from a host when you already know the host of the sample, therefore you can gain a run-time and computational advantage by removing these prior typically resource-heavy classification/profiling with more efficient methods. Furthermore, particularly with human samples, you can reduce the number of false positives during classification/profiling that occur due to host-sequence contamination in reference genomes on public databases.

> nf-core/taxprofiler currently offers host-removal via minimizer sketcing with Deacon for both short and long reads. For short reads, host removal can alternatively be performed via alignment against a reference genome using Bowtie2 or with Hostile that is using pre-defined indices. For long reads, minimap2 is offered as an alternative to Deacon. The filtered reads are then used for downstream classification/profiling.

Reviewer: ```suggestion
nf-core/taxprofiler currently offers host-removal via minimizer sketcing with Deacon for both short and long reads. For short reads, host removal can alternatively be performed via alignment against a reference genome using Bowtie2 or with Hostile that is using pre-defined indices. For long reads, hostile or minimap2 are offered as alternatives to Deacon. The filtered reads are then used for downstream classification/profiling.
```

Similarly to complexity filtering, host-removal can be useful for runtime optimisation and reduction in misclassified reads. It is not always necessary to report classification of reads from a host when you already know the host of the sample, therefore you can gain a run-time and computational advantage by removing these prior typically resource-heavy classification/profiling with more efficient methods. Furthermore, particularly with human samples, you can reduce the number of false positives during classification/profiling that occur due to host-sequence contamination in reference genomes on public databases.

> nf-core/taxprofiler currently offers host-removal via minimizer sketcing with Deacon for both short and long reads. For short reads, host removal can alternatively be performed via alignment against a reference genome using Bowtie2 or with Hostile that is using pre-defined indices. For long reads, hostile or minimap2 are offered as alternatives to Deacon. The filtered reads are then used for downstream classification/profiling.

Reviewer: ```suggestion
nf-core/taxprofiler currently offers host-removal via minimizer sketching with Deacon for both short and long reads. For short reads, host removal can alternatively be performed via alignment against a reference genome using Bowtie2 or with Hostile that is using pre-defined indices. For long reads, hostile or minimap2 are offered as alternatives to Deacon. The filtered reads are then used for downstream classification/profiling.
```

> For Deacon, Bowtie2 and minimap2, you can supply your reference genome in FASTA format with `--hostremoval_reference`. You can also optionally supply a directory containing pre-indexed Deacon/Bowtie2 index files with `--shortread_hostremoval_index` or a minimap2 `.mmi` file for `--longread_hostremoval_index`, however nf-core/taxprofiler will generate these for you if necessary. Pre-supplying the index directory or files can greatly speed up the process, and these can be re-used. For Hostile, you specifically need a name of the reference to either download or find the correct files in the index directory.

Reviewer: ```suggestion
For Deacon, Bowtie2, and minimap2, you can supply your reference genome in FASTA format with `--hostremoval_reference`. You can also optionally supply a directory containing pre-indexed Deacon/Bowtie2 index files with `--shortread_hostremoval_index` or a minimap2 `.mmi` file for `--longread_hostremoval_index`, however nf-core/taxprofiler will generate these for you if necessary. Pre-supplying the index directory or files can greatly speed up the process, and these can be re-used. For Hostile, you specifically need a name of the reference to either download or find the correct files in the index directory.
```
